Source/Urho3D/Network/Connection.cpp

```diff
@@ -288,15 +290,23 @@ void Connection::SendRemoteEvents()
     {
         statsTimer_.Reset();
         char statsBuffer[256];
-        sprintf(statsBuffer, "RTT %.3f ms Pkt in %d Pkt out %d Data in %.3f KB/s Data out %.3f KB/s", GetRoundTripTime(),
-            (int)0, //TODO: Packets in per second
-            (int)0, //TODO: Packets out per second
-            0.0f, //TODO: Bytes in per second
-            0.0f); // TODO: Bytes out per second
+        sprintf(statsBuffer, "RTT %.3f ms Pkt in %i Pkt out %i Data in %.3f KB/s Data out %.3f KB/s, Last heard %u", GetRoundTripTime(),
+            GetPacketsInPerSec(),
+            GetPacketsOutPerSec(),
+            GetBytesInPerSec(),
+            GetBytesOutPerSec(),
+            GetLastHeardTime());
         URHO3D_LOGINFO(statsBuffer);
     }
 #endif
 
+    if (packetCounterTimer_.GetMSec(false) > 1000)

```

Comment:
I feel strange to find the logic to be inside the `SendRemoteEvents()`. What if there is no remote events being sent for more than a second?

Reply:
SendRemoteEvents is always called on network update, the next few lines after this actually check if there are remote events that needs to be sent out, if not, the method exits.